### PR TITLE
`Form::TextInput` - Add loading state on "search" type

### PR DESCRIPTION
**Pull request by alex-ju**
### :pushpin: Summary

Add `@isLoading` on `<TextInput @type="search"/>`

### :hammer_and_wrench: Detailed description

When a search operation does not happen instantly it is often helpful to reflect the loading state in the search input.

We're introducing the `@isLoading` argument on `<TextInput @type="search"/>` to replace the magnifying glass with a loading spinner to be used while fetching the results in a search operation.

Note: we traditionally apply the animation to the loading icon via CSS, but in the current implementation, the magnifying glass is a background image on the `<input>` element, so it wouldn't work. For this reason, we create an SVG animation.

[Preview showcase](https://hds-showcase-6p7hxb1h8-hashicorp.vercel.app/components/form/text-input)
[Preview docs](https://hds-website-git-alex-ju-add-is-loading-to-sear-3cf15c-hashicorp.vercel.app/components/form/text-input?tab=code#formtextinputbase-1)

### :link: External links

Jira ticket: [HDS-2116](https://hashicorp.atlassian.net/browse/HDS-2116)
Figma: https://www.figma.com/file/noyY6dUMDYjmySpHcMjhkN/branch/905U1TC13EGOlA1DBNHrT1/HDS-Product---Components?type=design&mode=design&t=mvoLMF4ztq7moXzs-1

***

### 👀 Reviewer's checklist:

- [x] +1 Percy if applicable
- [x] Confirm that PR has a changelog update via [Changesets](https://github.com/changesets/changesets) if needed

:speech_balloon: Please consider using [conventional comments](https://conventionalcomments.org/) when reviewing this PR.

[HDS-2116]: https://hashicorp.atlassian.net/browse/HDS-2116?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by cveigt — CHANGES_REQUESTED** [review]
This looks great, Alex. Thanks for making these updates. I only have a piece of feedback. Can we change the color of the `loading` icon to match the `search` icon? It should be foreground faint (or the equivalent on the form tokens)

**Comment by alex-ju** [comment]
> This looks great, Alex. Thanks for making these updates. I only have a piece of feedback. Can we change the color of the `loading` icon to match the `search` icon? It should be foreground faint (or the equivalent on the form tokens)

Thanks for the review, Christian! Updated now to use foreground faint

File: .changeset/bright-cougars-dress.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"@hashicorp/design-system-components": minor
+---
+
+Add loading state on search input
```
Comment (didoo) [inline .changeset/bright-cougars-dress.md:5]: I think we should start to do something like this for our changelogs to they're easer to visually scan/parse (look at [last release](https://github.com/hashicorp/design-system/releases/tag/%40hashicorp%2Fdesign-system-components%402.7.1) to see what I mean)

```suggestion
`Form::TextInput` - Add loading state on "search" type
```

File: .changeset/tricky-badgers-try.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+"@hashicorp/design-system-tokens": minor
+---
+
+Add design token for loading state on search input
```
Comment (didoo) [inline .changeset/tricky-badgers-try.md:5]:
```suggestion
Add design token for loading state icon on search input
```

**Review by didoo — APPROVED** [review]
All good for me. Left only minor comments/suggestions.

**Review by cveigt — APPROVED** [review]

**Pull request merged**